plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/ray_aws_config.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from dataclasses import dataclass, field
+from typing import Any, Dict, List
+
+from hydra.core.config_store import ConfigStore
+from hydra.types import ObjectConf
+
+
+def default_field(obj):
+    return field(default_factory=lambda: obj)
+
+
+# This class maps to a Ray cluster config. To get more info on each field, please check:
+# https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml
+# Note: You cannot add arbitrary field not supported by Ray in this conf.
+@dataclass
+class RayClusterConfig:
```

Comment:
this is complex enough that you might want to the data in a yaml file and just the schema in the python file.
see the config of Hydra itself (and see the [this](https://hydra.cc/docs/next/tutorials/structured_config/schema))

We can discuss the advantages when we talk.